arch/xtensa/include/xtensa_asm2_s.h

```diff
@@ -9,6 +9,8 @@
 
 #include <zsr.h>
 #include "xtensa_asm2_context.h"
+#include <xtensa/config/tie.h>
+#include <xtensa/config/tie-asm.h>
```

Comment:
One of the (admittedly unclear and debatable) requirement behind #67513 was NOT to include .h files from the HAL and NOT get `XCHAL_*_HIFIn` macros from it.  Duplicate the HAL macros in Kconfig instead.

If you're including files from the HAL, then do you really need #67513 as is? Or a much smaller and more dynamic version of it?